### PR TITLE
[DO-NOT-MERGE] WIP fix for #1357 for reviewing

### PR DESCRIPTION
**Pull request by bbolli**

### CODE REVIEW

**Comment by bbolli** [comment]
Alright, I'll try to come up with a new series.

**Comment by dscho** [comment]
Thanks!

Please just force-push your branch once ready, this will update the PR as well.

**Comment by dscho** [comment]
@bbolli do you need anything from my side, or are you simply pressed for time (as we all are)?

**Comment by bbolli** [comment]
@dscho Actually I was on vacation, but I'll put in a few hours this weekend. Stay tuned...

**Comment by dscho** [comment]
> Stay tuned...

Thanks!

**Comment by dscho** [comment]
> Maybe we can introduce a shell function to facilitate all kinds of SDK things, including generating the desktop shortcut?

@bbolli what do you think of this `sdk` shell function idea?

**Comment by bbolli** [comment]
I'm in the process of implementing it. What other features should it have besides those in your sample?

**Comment by dscho** [comment]
> I'm in the process of implementing it.

Great!

> What other features should it have besides those in your sample?

I think we will find more and more things we will want to do, on the go.

I *could* imagine for example that it would be nice if there was an easy way to build a custom installer pulling in a git-extra from PR https://github.com/git-for-windows/build-extra/pull/171 and Git from PR https://github.com/git-for-windows/git/pull/1492 ;-)

But I think that it would be good to just have something for starters, and not block this here PR for possible crazy feature requests ;-)

File: git-extra/git-sdk.sh [hunk 1]
```diff
@@ -20,3 +20,52 @@ case "$rootdir" in
 	fi
 	;;
 esac
+
+
+sdk () {
+    case "$1" in
+    help|--help|-h)
+        echo "The '$0' shell function helps you to get up and running
```
Comment (dscho) [inline git-extra/git-sdk.sh]: Maybe `echo >&2 ...`? I am quite used to help messages going to stderr...

Comment (dscho) [reply git-extra/git-sdk.sh]: Or even `cat >&2 <<-EOF` (which has the advantage that the here-doc can be indented).

File: git-extra/git-sdk.sh [hunk 2]
```diff
@@ -20,3 +20,52 @@ case "$rootdir" in
 	fi
 	;;
 esac
+
+
+sdk () {
```
Comment (dscho) [inline git-extra/git-sdk.sh]: I think we only need one empty line before this function. What do you think?

File: git-extra/git-sdk.sh [hunk 3]
```diff
@@ -20,3 +20,54 @@ case "$rootdir" in
...
+	init)
+		case "$2" in
+		build-extra|git|MINGW-packages|MSYS2-packages)
+			dest="/usr/src/$2"
+			if test ! -d "$dest"/.git
+			then
+				mkdir -p "$dest" &&
+				git init "$dest" &&
+				git -C "$dest" remote add origin https://github.com/git-for-windows/$2
+			fi &&
```
Comment (dscho) [inline git-extra/git-sdk.sh]: I think I'll want to auto-initialize those worktrees, *without* actually fetching them (as a `git pull` will then Just Work).

**Comment by dscho** [comment]
@bbolli I added a few commits and force-pushed to your branch; I hope you do not mind...

Could you have a look and tell me what you think?

File: git-extra/git-sdk.sh [hunk 4]
```diff
@@ -45,23 +45,25 @@ sdk () {
...
+				https://github.com/git-for-windows/"$2" ||
+			sdk die "Could not initialize /usr/src/$2"
 			;;
 		*)
 			sdk die "Unhandled repository: $2" >&2
 			;;
 		esac
 		;;
+	init)
+		init_lazy "$2" &&
```
Comment (bbolli) [inline git-extra/git-sdk.sh]: I don't see `init_lazy` defined anywhere. Did you mean `sdk init-worktree`?

Comment (dscho) [reply git-extra/git-sdk.sh]: Whoops! You're right, I wanted to rename init-worktree to init-lazy...

**Comment by dscho** [comment]
@bbolli I would like to merge this as soon as possible, no idea what your availability is these days? I guess I'll just wait a couple minutes and then merge, build a new `git-extra` package, upload it to the Pacman repositories, and update the Git mirrors of the SDK so that we finally have something usable.

**Comment by dscho** [comment]
@bbolli I hope you don't mind that I merge this already ;-)

**Comment by dscho** [comment]
To clarify: I think it was most important to get this into the SDK, and we can easily develop it further from here. Thank you @bbolli for getting the ball rolling!

**Pull request merged**